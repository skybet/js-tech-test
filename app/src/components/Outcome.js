import React, { useContext } from "react";
import { useOutcome } from "../hooks/";
import { OddsContext } from "../App";

const Outcome = ({ id }) => {
  const [oddsType] = useContext(OddsContext);
  const outcome = useOutcome(id);

  if (!outcome) return <div>Loading...</div>;

  const {
    name,
    price: { num, den, decimal }
  } = outcome;

  return (
    <div className="c-outcome">
      <span className="u-clr--blue u-mr--100">{name}</span>
      <span className="u-clr--red">
        {oddsType === "fractional"
          ? `${num}/${den}`
          : Number(decimal).toFixed(2)}
      </span>
    </div>
  );
};

export default Outcome;
