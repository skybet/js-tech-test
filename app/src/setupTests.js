// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import { OddsContext } from "./App";

jest.mock("./hooks/useMarket", () =>
  jest.fn(() => ({
    marketId: 111,
    name: "Both teams to score",
    outcomes: [111, 222]
  }))
);

jest.mock("./hooks/useOutcome", () =>
  jest.fn(() => ({
    name: "Yes",
    price: {
      num: 3,
      den: 1,
      decimal: 4
    }
  }))
);

jest.mock("./hooks/useLiveEvents", () =>
  jest.fn(() => [
    {
      eventId: 111,
      name: "Everton vs West Ham",
      markets: [111, 222],
      competitors: [{ name: "Everton" }, { name: "West Ham" }],
      scores: { home: 3, away: 0 }
    }
  ])
);

export const renderWithContext = (component, oddsType = "fractional") =>
  render(
    <OddsContext.Provider value={[oddsType, jest.fn()]}>
      {component}
    </OddsContext.Provider>
  );
