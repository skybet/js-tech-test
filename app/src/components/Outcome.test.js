import React from "react";
import { renderWithContext } from "../setupTests";
import Outcome from "./Outcome";

describe("<Outcome />", () => {
  test("renders fractional odds as expected", async () => {
    const { asFragment } = renderWithContext(<Outcome />, "fractional");
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders decimal odds as expected", async () => {
    const { asFragment } = renderWithContext(<Outcome />, "decimal");
    expect(asFragment()).toMatchSnapshot();
  });

  test("falls back to decimal if bad data", async () => {
    const { asFragment } = renderWithContext(<Outcome />, "xyz");
    expect(asFragment()).toMatchSnapshot();
  });
});
