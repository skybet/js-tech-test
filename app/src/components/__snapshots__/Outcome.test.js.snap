// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Outcome /> falls back to decimal if bad data 1`] = `
<DocumentFragment>
  <div
    class="c-outcome"
  >
    <span
      class="u-clr--blue u-mr--100"
    >
      Yes
    </span>
    <span
      class="u-clr--red"
    >
      4.00
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Outcome /> renders decimal odds as expected 1`] = `
<DocumentFragment>
  <div
    class="c-outcome"
  >
    <span
      class="u-clr--blue u-mr--100"
    >
      Yes
    </span>
    <span
      class="u-clr--red"
    >
      4.00
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Outcome /> renders fractional odds as expected 1`] = `
<DocumentFragment>
  <div
    class="c-outcome"
  >
    <span
      class="u-clr--blue u-mr--100"
    >
      Yes
    </span>
    <span
      class="u-clr--red"
    >
      3/1
    </span>
  </div>
</DocumentFragment>
`;
